### PR TITLE
imp: separate env set up logs from task logs

### PR DESCRIPTION
**Pull request by alexjuda**
# The problem

Users fetch can workflow logs from CE. We expose it to the user as `sdk.WorkflowRun.get_logs()`. https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/187 changed the output type to a dataclass with `per_task` and `env` fields, but all logs were put into the `per_task` category.

# This PR's solution

* Splits the logs into `per_task` and `env` categories.
* Populates the `WorkflowLogs.env` field when accessing logs from CE.
* Local runtime and QE logs are unaffected for now.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_base/_logs/_regrouping.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+We want to use similar rules for parsing both CE responses and local runtime logs.
+"""
+import re
+
+# Example:
+# "/tmp/ray/session_latest/logs/worker-903b5bfa866935b17bbd3d3a64d22c7c6b746c2d522ad5fca9cf1f6a-01000000-249.err"
+WORKER_FILE_PATTERN = re.compile(r".+/worker-(\w+)-(\d+)-\d+.(\w{3})")
+
+# Example: "/tmp/ray/session_latest/logs/runtime_env_setup-01000000.log"
+ENV_SETUP_FILE_PATTERN = re.compile(r".+/runtime_env_setup-(\d+).log")
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_logs/_regrouping.py:17]: Note: this compile will happen at import time

File: src/orquestra/sdk/_base/_driver/_models.py [hunk 2]
```diff
@@ -358,3 +358,73 @@ class ProjectDetail(CommonResourceMeta, ResourceIdentifier):
...
+    """
+
+    timestamp: float
+    """
+    Unix timestamp in seconds with fraction for the moment when a log line is exported
+    from Ray system to Orquestra. It does not necessarily correspond to the particular
+    time that the message is logged by Ray runtime.
+    """
+
+    message: Message
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_driver/_models.py:427]: todo: add docstring

File: src/orquestra/sdk/_base/_driver/_ce_runtime.py [hunk 3]
```diff
@@ -448,9 +449,27 @@ def get_workflow_logs(self, wf_run_id: WorkflowRunId) -> WorkflowLogs:
...
+                task_logs.append(m.log)
+            elif _regrouping.ENV_SETUP_FILE_PATTERN.match(m.ray_filename) is not None:
+                env_logs.append(m.log)
+            else:
+                # Reasons for the "other" logs: future proofness and empathy. The server
+                # might return events from more files in the future. We want to let the
+                # user see it even this version of the SDK doesn't know how to
+                # categorize it. Noisy data is better than no data when the user is
+                # trying to find a bug.
+                other_logs.append(m.log)
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_driver/_ce_runtime.py:467]: todo: fix ordering of the "other" messages

File: tests/sdk/v2/driver/resp_mocks.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# noqa: E501
```
Comment (alexjuda) [inline tests/sdk/v2/driver/resp_mocks.py:1]: todo: leftover

File: src/orquestra/sdk/_base/_driver/_client.py [hunk 5]
```diff
@@ -657,23 +658,19 @@ def get_workflow_run_logs(self, wf_run_id: _models.WorkflowRunID) -> List[str]:
         decoded = untarred.read().decode()
 
         # Parse the decoded data as logs
-        # TODO: index by taskinvocationID rather than workflowrunID [ORQSDK-777]
-        logs = []
-        for section in decoded.split("\n"):
-            if len(section) < 1:
+        # TODO: index by taskinvocationID rather than workflowrunID [ORQSDK-840]
+        # logs = []
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_driver/_client.py:662]: **nitpick:** ‏commented out code

File: tests/sdk/v2/driver/data/get_wf_logs_response/messages.py [hunk 6]
```diff
@@ -0,0 +1,1869 @@
+# flake8: noqa
```
Comment (jamesclark-Zapata) [inline tests/sdk/v2/driver/data/get_wf_logs_response/messages.py:1]: I made a similar comment on Benjamin's PR.

Do we need to compare directly against all these messages? or can we compare on properties of the messages and look for sentinal values.

Having so many log lines will be annoying to update.

Comment (alexjuda) [reply tests/sdk/v2/driver/data/get_wf_logs_response/messages.py:1]: I wasn't entirely happy with my approach either. I'll see how property-based assertions would play out for is.

Comment (alexjuda) [reply tests/sdk/v2/driver/data/get_wf_logs_response/messages.py]: 71593f0 WDYT

**Review by jamesclark-Zapata — APPROVED** [review]

File: tests/sdk/v2/driver/data/get_wf_logs_response/messages.py (hunk 6 above)
Comment (jamesclark-Zapata) [reply tests/sdk/v2/driver/data/get_wf_logs_response/messages.py]: I think thats nicer to work with in the future 👍

**Comment by jamesclark-Zapata** [comment]
Merging on behalf of @alexjuda to unblock #195

**Pull request merged**